deno_dist/types.ts

```diff
@@ -1658,44 +1658,50 @@ type ExtractParams<Path extends string> = string extends Path
 type FlattenIfIntersect<T> = T extends infer O ? { [K in keyof O]: O[K] } : never
 
 export type MergeSchemaPath<OrigSchema extends Schema, SubPath extends string> = Prettify<{
-  [P in keyof OrigSchema as MergePath<SubPath, P & string>]: {
-    [M in keyof OrigSchema[P]]: OrigSchema[P][M] extends {
-      input: infer Input
-      output: infer Output
-      outputFormat: infer OutputFormat
-      status: infer Status
-    }
-      ? {
-          input: Input extends { param: infer _ }
-            ? ExtractParams<SubPath> extends never
-              ? Input
-              : FlattenIfIntersect<
-                  Input & {
-                    param: {
-                      // Maps extracted keys, stripping braces, to a string-typed record.
-                      [K in keyof ExtractParams<SubPath> as K extends `${infer Prefix}{${infer _}}`
-                        ? Prefix
-                        : K]: string
-                    }
-                  }
-                >
-            : RemoveBlankRecord<ExtractParams<SubPath>> extends never
-            ? Input
-            : Input & {
-                // Maps extracted keys, stripping braces, to a string-typed record.
+  [P in keyof OrigSchema as MergePath<SubPath, P & string>]: OrigSchema[P] extends {
+    [M in keyof OrigSchema[P]]: OrigSchema[P][M]
+  }
+    ? {
+        [M in keyof OrigSchema[P]]: MergeEndpointParamsWithPath<OrigSchema[P][M], SubPath>
+      }
+    : never
+}>
```

Comment:
Could we make this part and just put
```
Prettify<{
  [P in keyof OrigSchema as MergePath<SubPath, P & string>]: {
    [M in keyof OrigSchema[P]]: MergeEndpointParamsWithPath<OrigSchema[P][M], SubPath>
  }
}>
```

What do you think?

Reply:
You are right! I've fixed it.